src/disco/bank/fd_txncache.h

```diff
@@ -197,6 +197,7 @@ struct fd_txncache_insert {
   uchar const * blockhash;
   uchar const * txnhash;
   ulong         slot;
+  ulong         txnhash_offset;
```

Comment:
This doesn't make sense on the insert API, typically in the insert case the caller does not know the txnhash_offset.

You should add a separate snapshot_restore type functionality and pass a txnhash_offset only there.